src/inference/src/dev/core_impl.cpp

```diff
@@ -984,84 +932,204 @@ void ov::CoreImpl::AddExtensionUnsafe(const InferenceEngine::IExtensionPtr& exte
+    std::lock_guard<std::mutex> lock(_CoreConfigCacheMutex);
+    if (!device_name.empty()) {
+        if (_core_properties_cache_per_device.count(device_name) > 0) {
+            auto& cache_config = _core_properties_cache_per_device.at(device_name);
+            auto item = cache_config._properties.find(config_name);
+            if (item != cache_config._properties.end()) {
+                return item->second;
+            }
+        }
     }
 
-    it = config.find(ov::hint::allow_auto_batching.name());
-    if (it != config.end()) {
-        auto flag = it->second.as<bool>();
-        flag_allow_auto_batching = flag;
-        config.erase(it);
+    auto item = _core_plugin_properties._properties.find(config_name);
+    if (item != _core_plugin_properties._properties.end()) {
+        return item->second;
+    }
+
+    item = _core_global_properties._properties.find(config_name);
+    if (item != _core_global_properties._properties.end()) {
+        if (config_name == ov::force_tbb_terminate.name()) {
+            const auto flag = InferenceEngine::executorManager()->getTbbFlag();
+            return flag ? CONFIG_VALUE(YES) : CONFIG_VALUE(NO);
+        }
+        return item->second;
+    }
+
+    IE_THROW() << "Exception: unsupported core property name: '" << config_name << "'";
+}
+
+void ov::CoreImpl::CoreConfig::print_core_properties() const {
+    // std::lock_guard<std::mutex> lock(_CoreConfigCacheMutex);
+    std::cout << "Default core property:" << std::endl;
+    for (auto& it : _core_plugin_properties._properties) {
+        std::cout << "\t" << it.first.c_str() << " : " << it.second.as<std::string>().c_str() << std::endl;
+    }
+    std::cout << "\t_cacheManager = " << _core_plugin_properties._cacheManager << std::endl;
+
+    for (auto& dev : _core_properties_cache_per_device) {
+        std::cout << dev.first.c_str() << " core property:" << std::endl;
+        for (auto& it : dev.second._properties) {
+            std::cout << "\t" << it.first.c_str() << " : " << it.second.as<std::string>().c_str() << std::endl;
+        }
+        std::cout << "\t_cacheManager = " << dev.second._cacheManager << std::endl;
+    }
+    std::cout << std::endl;
+}
+
+bool ov::CoreImpl::CoreConfig::is_core_config(const std::string& config_name) const {
+    auto ret = _core_plugin_properties._properties.find(config_name) != _core_plugin_properties._properties.end() ||
+               _core_global_properties._properties.find(config_name) != _core_global_properties._properties.end();
+    return ret;
+}
+
+bool ov::CoreImpl::CoreConfig::core_config_supported(const ov::Plugin& plugin, const std::string& config_name) const {
+    auto supported = false;
+    auto device_name = plugin.get_name();
+
+    if (_core_plugin_properties._properties.find(config_name) != _core_plugin_properties._properties.end()) {
+        supported = device_name.find("CPU") != std::string::npos || device_name.find("GPU") != std::string::npos ||
```

Comment:
From my point of view Core should know nothing about plugins. It is a bad design that now it is not true, but I don't want to extend these practices to new places.

Reply:
I don't fully understand maybe the idea, but it is not clear for me, why plugins should change something for this change?
Why cannot it be handled on Core level?

Reply:
The main idea is to send all core properties to plugin by compile_model() method, while plugin decides how to handle these core properties, maybe ignore or remove unused core properties.
But currently plugins don't have the same process logic for properties, so we need change plugins to let them handle core properties correctly. Once is done, in the future, plugin will not changes if we add new core properties.